dotCMS/src/main/java/com/dotmarketing/common/db/DotConnect.java

```diff
@@ -770,6 +711,83 @@ private void executeQuery(Connection conn) throws SQLException {
 
     }
 
+    /**
+     * Sets the results and objectResults from a given ResultSet
+     *
+     * @param rs
+     * @throws SQLException
+     */
+    public void fromResultSet(ResultSet rs) throws SQLException {
+
+        results = new ArrayList<Object>();
+        objectResults = new ArrayList<Map<String, Object>>();
+        gotResult = true;
+
+        if (rs != null) {
+
+            var rsmd = rs.getMetaData();
+
+            // move to the starter row
+            for (int i = 0; i < startRow; i++) {
+                rs.next();
+            }
+
+            int i = 0;
+
+            while (rs.next() && (maxRows <= 0 || i < maxRows)) {
+                HashMap<String, String> vars = new HashMap<String, String>();
+                HashMap<String, Object> objvars = new HashMap<String, Object>();
+
+                for (int j = 1; j <= rsmd.getColumnCount(); j++) {
+                    String x = rsmd.getColumnLabel(j) + "";
+                    try {
+                        if ((rs.getString(x) == null) || rs.getString(x).equals("null")) {
+                            x = x.toLowerCase();
```

Comment:
encapsulate this in a method

Reply:
This is a good suggestion Jonathan, I had to refactor the `executeQuery` method to be able to re-use the ResulSet parsing from a sql.Statement response, but we want to touch as little as possible the DotConnect class, I was recommended to leave it as it is.